### ISSUE TITLE
Config-driven Weighted Loss Function (Emphasize Q₀)

### ISSUE THREAD

**Issue by hmehryar** [hmehryar/ml-jet-param-predictor#27]
## Why
We already classify **Energy-loss** and **α_s** with high confidence.  
The weaker task is **Q₀**, so we want to emphasize it in training by up-weighting its loss term.  
This allows our best ViT model to focus more on improving Q₀ classification accuracy.

## What
- Extend YAML config schema to include `loss.weights`.
- Parse weights in `config.py` and expose them as `cfg.loss_weights`.
- Modify `train_one_epoch(...)` in `train_epoch.py` to use weighted total loss (λ₁, λ₂, λ₃).
- Pass `cfg.loss_weights` from `training_loop.py` into both `train_one_epoch(...)` and `evaluate(...)`.
- Update `evaluate(...)` to compute weighted total loss for consistency, while still logging per-head losses.
- Keep schedulers/early stopping unchanged (still use total accuracy).
- Defaults should be backward compatible (`1.0` for all weights).

## Acceptance Criteria
- Config allows values like:
  ```yaml
  loss:
    weights:
      energy_loss_output: 1.0
      alpha_output: 1.0
      q0_output: 3.0 

- Training and evaluation report:

 - loss = weighted sum (e.g., Q₀ weighted ×3.0)

 - loss_energy, loss_alpha, loss_q0 remain unweighted for monitoring.

- Resume and checkpointing still work without changes.

- Logs clearly reflect when Q₀ weighting is applied.
